apisix/discovery/nacos/init.lua

```diff
@@ -54,6 +55,23 @@ local function get_key(namespace_id, group_name, service_name)
     return namespace_id .. '.' .. group_name .. '.' .. service_name
 end
 
+
+local function metadata_contains(host_metadata, route_metadata)
+    if not route_metadata or not next(route_metadata) then
+        return true
+    end
+    if not host_metadata or not next(host_metadata) then
+        return false
+    end
+
+    for k, v in pairs(route_metadata) do
+        if host_metadata[k] ~= v then
+            return false
+        end
+    end
+    return true
```

Comment:
It seems that there is a problem with the logic here:

<img width="755" height="300" alt="Image" src="https://github.com/user-attachments/assets/0257ee6e-20b9-4f93-bda3-77d97bac8eb7" />

`route_metadata` will not be nil or empty